regression/tests/enterprise/test_enterprise_data_consent.py

```diff
@@ -0,0 +1,126 @@
+"""
+Enterprise Data consent tests
+"""
+from regression.pages.enterprise.enterprise_data_consent_page import (
+    EnterpriseDataConsentPage
+)
+from regression.pages.whitelabel.basket_page import SingleSeatBasketPage
+from regression.tests.enterprise.enterprise_test_base import EnterpriseTestBase
+
+
+class TestEnterpriseDataConsent(EnterpriseTestBase):
+    """
+    Test Enterprise Data Consent
+    """
+
+    CONSENT_MSG = "To access this course, you must first consent to share"
+    CONSENT_DECLINE_WARNING = "We could not enroll you"
+
+    def setUp(self):
+        """
+        Initialize all page objects
+        """
+        super(TestEnterpriseDataConsent, self).setUp()
+        self.browser.maximize_window()
+        self.enterprise_data_consent = EnterpriseDataConsentPage(self.browser)
+
+    def test_data_consent_page_details(self):
+        """
+        Scenario: To verify that user sees the correct settings on data consent
+            page
+            Given that a new user registers for the course through enterprise
+            When this user clicks next from enrollment page
+            Then this user is taken to data consent page
+            And relevant details are shown on this page
+        """
+        # The edX site is visited just to make sure that when user jumps to
+        # edX from portal we don't have to handle authentication popup
+        self.lms_login.visit()
+        # Enterprise portal flow
+        self.login_to_enterprise_portal()
+        self.access_course()
+        self.register_enterprise_edx_user()
+        # Verify that user is on course enrollment page and correct course
+        # is displayed there
+        self.enterprise_course_enrollment.wait_for_page()
+        self.assertDictEqual(
+            self.ENT_COURSE_TITLE,
```

Comment:
We are doing all the login stuff and going to enterprise course enrollment page in lot of other places. I would suggest creating a helper method like `go_to_enterprise_course_enrollment_page` and move all this logic there.